### PR TITLE
Moved step of building static assets to dockerfile

### PR DESCRIPTION
**Pull request by bartsmykla**
Closes: https://github.com/openfaas/openfaas-cloud/issues/325

Signed-off-by: Bart Smykla <bsmykla@vmware.com>

## Description

- Changed dashboard-function type to dockerfile and moved building
  static files step into it. There is no need to have dist directory
  inside repository now

## How Has This Been Tested?

- Deployed dashboard to my cluster and tested if all features works
- Started development version of dashboard `yarn start` inside `dashboard/client` directory checking if everything works for development purposes also

## How are existing users impacted? What migration steps/scripts do we need?
Users should not be affected because only building process of dashboard changed. The results should be the same.

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

**Comment by alexellis** [comment]
Thanks for getting to this.

What's the new instructions for local development vs publishing a container? Should we give some help to new contributors?

**Comment by bartsmykla** [comment]
Actually I removed mention of manually step of building it using `make build` from documentation, and nothing else changed. Now for deploying a container you can use just `faas up` and it will build everything and deploy automatically.

Local development didn't change at all. I just tested if everything about it works as it was before.

File: dashboard/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+### Final stage
+
+FROM node:10.12.0-alpine
+
+WORKDIR /root/
+
+RUN addgroup -S app && adduser -S -g app app
+
+RUN apk --no-cache add curl vim \
```
Comment (ewilde) [inline dashboard/Dockerfile]: is there a reason this `apk` install / `curl` isn't done in the previous build stage and then copied into this final build stage

**Comment by alexellis** [comment]
Please can you review the instructions for developers to make sure there is adequate help for getting started and running locally / in a Docker image.

**Comment by alexellis** [comment]
Needs a rebase please..

File: dashboard/Dockerfile (hunk 1 above)
Comment (alexellis) [reply dashboard/Dockerfile]: @bartsmykla WDYT?

Comment (bartsmykla) [reply dashboard/Dockerfile]: Hi @ewilde huge sorry, for not getting back to you. I'll respond to you today!

**Comment by bartsmykla** [comment]
It's rebased now

**Comment by bartsmykla** [comment]
But I need to test it again, after rebasing with master

**Comment by alexellis** [comment]
(OK so this is with you for now.)

**Comment by bartsmykla** [comment]
@alexellis rebased again, and tested. I'm confident that it's working as expected

File: dashboard/Dockerfile (hunk 1 above)
Comment (bartsmykla) [reply dashboard/Dockerfile]: @ewilde I just understood your point, and I fixed that. Thank you, and sorry for getting back to this so late

**Comment by bartsmykla** [comment]
Derek remove label: blocked

**Comment by alexellis** [comment]
@Waterdrips do you feel we could we benefit from continuing to implement this change? The primary difference is that the assets are not generated on disk and built into a container, they're built directly within the container we ship.

**Comment by Waterdrips** [comment]
So I personally feel that adding over 150 lines just to remove the need to run `make && faas-cli up` instead of `faas-cli up` isnt worth it.

Plus we then have to maintain these templates rather than just change/upgrade using one of the other openfaas templates

**Pull request closed without merging**